### ISSUE TITLE
CuDNN batchnorm has batch size limit

### ISSUE THREAD

**Issue by soumith** [pytorch/pytorch#2917]
```python
import torch
import torch.nn as nn
from torch.autograd import Variable

torch.backends.cudnn.enabled=True
x = Variable( torch.rand(140000,1).contiguous()).cuda()

print (torch.backends.cudnn.version())

bn = nn.BatchNorm1d(1)
bn.cuda()

xbn = bn(x)
xbn.size()
```

This will give CUDNN_STATUS_NOT_SUPPORTED error, as reported here: https://discuss.pytorch.org/t/cudnn-status-not-supported-with-large-bach-size-when-using-batchnorm/7014/4

**Comment by soumith** [comment]
The magic number is `131070`. Anything above this batch size fails.
